### PR TITLE
remove Godep, add vendor.json and update CONTRIBUTING.md

### LINKED ISSUE #2221: Swarm fails to find packages when running unit testing with `go test -race ./...`!
Hi, All,

When running unit tests locally on my own machine, it always reports error like these:
```
root@ubuntu:~/gocode/src/github.com/docker/swarm# go test -race ./...
vendor/github.com/Sirupsen/logrus/examples/hook/hook.go:5:2: cannot find package "gopkg.in/gemnasium/logrus-airbrake-hook.v2" in any of:
	/root/gocode/src/github.com/docker/swarm/vendor/gopkg.in/gemnasium/logrus-airbrake-hook.v2 (vendor tree)
	/usr/local/go/src/gopkg.in/gemnasium/logrus-airbrake-hook.v2 (from $GOROOT)
	/root/gocode/src/gopkg.in/gemnasium/logrus-airbrake-hook.v2 (from $GOPATH)
```

missing packages in vendor, GOROOT and GOPATH.

Maybe we need to add it in the /vendor, but now /vendor dir misses vendor.json, thus still broken.
https://github.com/docker/swarm/pull/2220 has tried to make vendor complete.

### PR DESCRIPTION
**Pull request by allencloud**
Actually, currently Swarm uses vendor to manage golang package dependencies, but not so completely.
Still missing vendor.json in /verndor.
And when running `go test -race ./...` to do the unit testing, there are still some packages missing in /vendor.
Without improving the vendor usage, no packages can be added to /vendor automatically.
 
This pr did;
1.remove Godeps dir when useing `govendor migrate`;
2.add vendor.json when using `govendor migrate`;
3.update CONTRIBUTING.md to replace  godeps with govendor.
4.change `go test ./...` to use the command in travis.yml.

This PR fixes #2221 .

Signed-off-by: allencloud <allen.sun@daocloud.io>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,9 +15,9 @@ Otherwise, go read Docker's
 
 ### Development Environment Setup
 
-Swarm is written in [the Go programming language](http://golang.org) and manages its dependencies using [Godep](http://github.com/tools/godep).  This guide will walk you through installing Go, forking the Swarm repo, building Swarm from source and contributing pull requests (PRs) back to the Swarm project.
+Swarm is written in [the Go programming language](http://golang.org) and manages its dependencies using [govendor](https://github.com/kardianos/govendor).  This guide will walk you through installing Go, forking the Swarm repo, building Swarm from source and contributing pull requests (PRs) back to the Swarm project.
```
Comment (dongluochen) [inline CONTRIBUTING.md]: It'd be better to provide version for the switch between Godep and Govendor. Users running older versions should check CONTRIBUTING.md in the source.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -217,19 +217,17 @@ You can also find maintainers' email addresses in `MAINTAINERS`.
...
-         # then manually add missing dependencies
+$GOBIN/govendor get <package_url>
+$GOBIN/govendor add <package_url>
 ```
 
 To make sure you newly added codes will make the build process happy, you can try building Swarm in the same way as defined in `Dockerfile`.
 
 ```sh
-go install
+go install
```
Comment (dongluochen) [inline CONTRIBUTING.md]: Remove empty space at the end of line.

**Comment by dongluochen** [comment]
Ping @vieux .

File: CONTRIBUTING.md (hunk 1 above)
Comment (allencloud) [inline CONTRIBUTING.md]: OK, I will add that soon.
Thanks @dongluochen

**Comment by allencloud** [comment]
ping @vieux 
Please take a look at this.
This pr completed the govendor thing.

**Comment by vieux** [comment]
NOT LGTM

I'm not in favor of this change. Godeps might have some flaws but it's well know and well maintained.
I don't know this govendor tool

ping @abronan

**Comment by abronan** [comment]
Sorry, missed that one. I would lean towards @vieux's opinion on this. Godeps is our historical dependency management tool in Swarm and I see no major reason to switch even though the experience can be frustrating at times, especially with the latest changes in vendoring. Let's keep Godeps for now and let's include the changes in the docs instead.

**Comment by abronan** [comment]
@allencloud thank you for the proposal, I'm going to close this one though as it seems that the consensus is to stick with Godep.

**Pull request closed without merging**